sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -200,8 +205,31 @@ public void recordOpenConnectionsAndInitCachesStarted(List<CosmosContainerIdenti
             }
         }
     }
+    
+    public void recordCollectionRidsAndUrisUnderOpenConnectionsAndInitCaches(String collectionRid, List<String> addressUrisAsString) {
+        this.collectionRidsAndUrisUnderOpenConnectionAndInitCaches.compute(collectionRid, (ignore, urisAsString) -> {
 
-    public Disposable getBackgroundOpenConnectionsPublisher() {
+            if (urisAsString == null) {
+                urisAsString = new HashSet<>(addressUrisAsString);
+            } else {
+                urisAsString.addAll(addressUrisAsString);
+            }
+
+            this.addressUrisUnderOpenConnectionsAndInitCaches.addAll(addressUrisAsString);
```

Comment:
same gap as existing issue today (like stop the proactive loop if the partition does not exists anymore) - if the addresses does not belong to the container anymore, there is no remove from the set. In the future when we solve it, this is another place to take care of

Reply:
Yeah this is in the works.